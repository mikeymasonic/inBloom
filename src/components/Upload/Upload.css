.Upload{
  /* background-color: rgba(133, 32, 119, 0.658); */
  iframe {    
    border: 0;
   }
}