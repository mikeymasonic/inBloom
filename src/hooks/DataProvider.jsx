import React, { createContext, useState, useContext } from 'react';
import PropTypes from 'prop-types';

const DataContext = createContext();

export const DataProvider = ({ children }) => {
  // const [song, setSong] = useState('../mp3/example2.mp3');
  const [song, setSong] = useState(null);
  const [mapLocation, setMapLocation] = useState(null);
  const [playerVisible, setPlayerVisible] = useState(false);
  const [visualizerDisplay, setVisualizerDisplay] = useState(false);

  const handleSong = (track) => {
    setSong(track);
    setMapLocation(track.mapLocation);
  };

  const handleMapLocation = (location) => {
    setMapLocation(location);
  };

  const handlePlayerVisible = (bool) => {
    setPlayerVisible(bool);
  };

  const handleVisualizerDisplay = (bool) => {
    setVisualizerDisplay(bool);
  };

  return (
    <DataContext.Provider
      value={{
        song,
        mapLocation,
        playerVisible,
        visualizerDisplay,
        handleSong,
        handleMapLocation,
        handlePlayerVisible,
        handleVisualizerDisplay,
      }}
    >
      {children}
    </DataContext.Provider>
  );
};

DataProvider.propTypes = {
  children: PropTypes.node,
};

// get state
export const useSong = () => {
  const { song } = useContext(DataContext);
  return song;
};

export const useMapLocation = () => {
  const { mapLocation } = useContext(DataContext);
  return mapLocation;
};

export const usePlayerVisible = () => {
  const { playerVisible } = useContext(DataContext);
  return playerVisible;
};

export const useVisualizerDisplay = () => {
  const { visualizerDisplay } = useContext(DataContext);
  return visualizerDisplay;
};

// get handlers
export const useHandleSong = () => {
  const { handleSong } = useContext(DataContext);
  return handleSong;
};

export const useHandleMapLocation = () => {
  const { handleMapLocation } = useContext(DataContext);
  return handleMapLocation;
};

export const useHandlePlayerVisible = () => {
  const { handlePlayerVisible } = useContext(DataContext);
  return handlePlayerVisible;
};

export const useHandleVisualizerDisplay = () => {
  const { handleVisualizerDisplay } = useContext(DataContext);
  return handleVisualizerDisplay;
};

