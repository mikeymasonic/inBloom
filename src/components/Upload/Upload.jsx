import React from 'react';
import { useMapLocation } from '../../hooks/DataProvider';
import PropTypes from 'prop-types';
import { uploads } from '../../utils/data';
import styles from './Upload.css';

const Upload = ({ close }) => {
  const mapLocation = useMapLocation();
  const uploadDescription = uploads[0].description;

  const uploadNodes = uploads.map((upload) => {
    if (mapLocation === upload.mapLocation) {
      return (
        <section 
          key={upload.mapLocation}
          className={styles.uploadContainer}>
          <h1>Upload for {upload.mapLocation}</h1>
          <iframe src={upload.url} width="90%"></iframe>
        </section>
      );
    }
  });

  return (
    <section className={styles.Upload}>
      <svg 
        className={styles.close} 
        fill="none" 
        stroke="currentColor" 
        viewBox="0 0 24 24" 
        xmlns="http://www.w3.org/2000/svg"
        height='1.5em'
        onClick={close}>
        <path 
          strokeLinecap="round" 
          strokeLinejoin="round" 
          strokeWidth={2} 
          d="M6 18L18 6M6 6l12 12" />
      </svg>
      {uploadNodes}
      <section className={styles.description}>
        {uploadDescription}
      </section>
    </section>
  );
};

Upload.propTypes = {
  close: PropTypes.func,
};

export default Upload;
