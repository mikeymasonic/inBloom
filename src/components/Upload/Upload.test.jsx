import React from 'react';
import { render, cleanup } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import { DataProvider } from '../../hooks/DataProvider';
import Upload from './Upload';

describe('Player component', () => {
  afterEach(() => cleanup());
  it('renders Player', () => {
    const { asFragment } = render(
      <Router>
        <DataProvider>
          <Upload />
        </DataProvider>
      </Router>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
