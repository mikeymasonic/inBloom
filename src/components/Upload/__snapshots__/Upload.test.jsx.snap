// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Player component renders Player 1`] = `
<DocumentFragment>
  <section
    class="Upload"
  >
    <svg
      class="close"
      fill="none"
      height="1.5em"
      stroke="currentColor"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M6 18L18 6M6 6l12 12"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
      />
    </svg>
  </section>
</DocumentFragment>
`;
